packages/appstore/src/modules/tour-guide/tour-guide.tsx

```diff
@@ -54,78 +44,96 @@ const TourGuide = () => {
-            <div
-                onClick={() => {
-                    trackStepForward(7);
-                }}
-            >
-                <Localize i18n_default_text='Next' />
-            </div>
-        );
+    tour_step_locale.next = (
+        <div
+            className='tour-guide__next'
+            onClick={() => {
+                if (joyride_index === 0) trackStepForward(7);
+                setJoyrideIndex(prev => prev + 1);
+            }}
+        >
+            <Localize i18n_default_text='Next' />
+        </div>
+    );
 
-        high_risk_tour_step_locale.next = (
-            <div
-                onClick={() => {
-                    trackStepForward(7);
-                }}
-            >
-                <Localize i18n_default_text='Next' />
-            </div>
-        );
-    }
+    high_risk_tour_step_locale.next = (
+        <div
+            className='tour-guide__next'
+            onClick={() => {
+                if (joyride_index === 0) trackStepForward(7);
+                setJoyrideIndex(prev => prev + 1);
+            }}
+        >
+            <Localize i18n_default_text='Next' />
+        </div>
+    );
+
+    tour_step_locale.back = (
+        <SpanButton
+            has_effect
+            secondary
+            medium
+            className='tour-guide__last'
+            onClick={() => {
+                setJoyrideIndex(prev => prev - 1);
+            }}
+        >
+            <Localize i18n_default_text='Back' />
+        </SpanButton>
+    );
 
     if (tour_step_config.length === joyride_index + 1) {
         tour_step_locale.back = (
             <SpanButton
+                className='tour-guide__back'
                 has_effect
                 text={localize('Repeat tour')}
                 secondary
                 medium
                 onClick={() => {
                     trackOnboardingRestart();
                     setIsFirstTimeVisit(false);
-                    history.push(routes.onboarding);
                     toggleIsTourOpen(true);
+                    setJoyrideIndex(0);
                 }}
             />
         );
     }
 
     high_risk_tour_step_locale.back = (
         <SpanButton
+            className='tour-guide__back'
             has_effect
             text={localize('Repeat tour')}
             secondary
             medium
             onClick={() => {
                 trackOnboardingRestart();
                 setIsFirstTimeVisit(false);
-                history.push(routes.onboarding);
                 toggleIsTourOpen(true);
+                setJoyrideIndex(0);
             }}
         />
     );
-
     const low_risk = content_flag === ContentFlag.LOW_RISK_CR_NON_EU || content_flag === ContentFlag.LOW_RISK_CR_EU;
-
+    if (!is_tour_open && !should_trigger_tour_guide) return null;
     return (
         <Joyride
-            run={(!is_onboarding_visited && is_tour_open) || should_trigger_tour_guide}
+            run={true}
```

Comment:
according to the joyride docs `run` is by default `true` 

ref: https://docs.react-joyride.com/props